src/Function.cpp

```diff
@@ -441,6 +460,18 @@ Function::make_random_signature(const CGContext& cg_context, const Type* type, c
 
 	if(CGOptions::func_attr_aligned() && rnd_flipcoin(FuncAttrAligned))
 		f->func_attr_aligned = true;
+
+	if(CGOptions::func_attr_flag()){
+		f->func_attr_flag = true;
+		string name = "hot";
+
+		FunctionAttribute *base;
```

Comment:
Only what you need here is `f->attributes.push_back(new BooleanFunctionAttribute(constructor arguments));`